src/exception/types.ts

```diff
@@ -118,10 +128,16 @@ export const isFollowupExceptionRecord = (input: any): input is FollowUpExceptio
 
 // type guards
 export const isEntityExceptionRecord = (input: any): input is EntityExceptionRecord => {
-  return (
-    isExceptionRecordCheck(input) &&
-    (isSpecimenExceptionRecord(input) || isFollowupExceptionRecord(input))
-  );
+  const hasDonorIdField =
+    'submitter_donor_id' in input && typeof input.submitter_donor_id === 'string';
+
+  if (hasDonorIdField && isExceptionRecord(input)) {
+    // remove based exception record fields to validate specific entity
+    const entityFields = _.omit(input, baseEntityExceptionFields);
+    // can't have more than one identifying field eg. submitter_specimen_id AND submitter_followup_id
+    return Object.keys(entityFields).length === 1;
```

Comment:
tricky. should work.

we should consider updating these to use a schema validator like zod sometime.

Reply:
yes. zod is looking great.